### DISCUSSION TITLE
Set the language selecting only a specific file.

### DISCUSSION THREAD

**Discussion by JoseDeFreitas** [github-linguist/linguist#5273] [Q&A] [answered]
Hello. I'm using `linguist-language` to change the language of a file, but it's not working. First, I set it to be `*.md` and it worked as expected. But then I changed it to `README.md` (so it only counts that file) but doesn't seem it's correct. I navigated through the documentation and found an example of changing the language of all files with a specific extension, but as you see I only want to set
that specific file to count it as the language I decide. Is it possible? Am I doing it wrong? I've waited some hours because it may be slow at changing the language but the language is still not the one I want.

**Comment by Nixinova** [comment, marked as answer]
`README.md linguist-language=Foo` should work; do note that Linguist caches language statistics so you may need to edit the readme file after changing the gitattributes to force Linguist to update its classification of that file.

**Reply by JoseDeFreitas** [reply]
Hello. I've tried doing what you wrote: updating the README.md file after setting the language of it in the .gitattributes file. However, some time has passed and it's not working... Both files are at the root project directory. The language I'm setting is C++ so my file is `README.md linguist-language=cpp`. I don't know if I have to wait longer. First time I did it (`*.md linguist-language=cpp`) to try it was updated just seconds after 🤔.

**Reply by lildude** [reply]
In the case of the `README.md` file, you need to use two overrides because it is also considered documentation:

https://github.com/github/linguist/blob/cf480b82c5860c9efe44a74da87503fbbecfec19/lib/linguist/documentation.yml#L29

So you'll need something like:

```gitattributes
README.md linguist-language=cpp -linguist-documentation
```

> First time I did it (`*.md linguist-language=cpp`) to try it was updated just seconds after 🤔.

This was likely picking up another file and not the `README.md` file.

**Reply by JoseDeFreitas** [reply]
Hi @lildude! I just made those changes and it's now working. I can see the language I want for only that file. Thank you!
